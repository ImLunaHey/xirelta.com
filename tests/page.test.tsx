import { expect, test } from 'bun:test';
import { Application } from 'xirelta';
import pretty from 'pretty';
import { randomUUID } from 'crypto';

test('page renders', async () => {
    const app = new Application({
        logger: {
            debug: () => {},
            info: () => {},
            error: () => {},
        }
    });
    const server = await app.start();

    const response = await fetch(`http://localhost:${server.port}/`).then(response => response.text());

    expect(response).toContain('Xirelta Web Framework');
    expect(pretty(response)).toMatchSnapshot();

    await app.stop();
});

test('404 renders', async () => {
    const app = new Application({
        logger: {
            debug: () => {},
            info: () => {},
            error: () => {},
        }
    });
    const server = await app.start();

    const response = await fetch(`http://localhost:${server.port}/${randomUUID()}`).then(response => response.text());

    expect(response).not.toContain('Xirelta Web Framework');
    expect(pretty(response)).toMatchSnapshot();

    await app.stop();
});
