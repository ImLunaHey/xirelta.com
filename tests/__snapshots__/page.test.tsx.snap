// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`page renders 1`] = `
"<html lang="en">

  <head>
    <meta charSet="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Xirelta</title>
    <script src="https://cdn.tailwindcss.com"></script>
  </head>

  <body class="bg-[#0d0e11] h-screen flex items-center justify-center">
    <main class="w-[500px] text-center">
      <header class="p-2 mb-[12px] rounded-[4px] border-[2px] border-gray-200 text-gray-400 text-center"><a href="https://github.com/ImLunaHey/xirelta"><img src="https://raw.githubusercontent.com/ImLunaHey/xirelta/main/.github/assets/logo.png" alt="Xirelta Logo" /></a><a href="https://github.com/ImLunaHey/xirelta">
          <h1>Xirelta Web Framework</h1>
        </a></header>
      <section class="p-2 mb-[12px] rounded-[4px] border-[2px] border-gray-200 text-gray-400">
        <h2>Installation</h2><pre><code>bun add xirelta</code></pre>
      </section>
      <footer class="p-2 mb-[12px] rounded-[4px] border-[2px] border-gray-200 text-gray-400">
        <p>License: MIT</p>
      </footer>
    </main>
  </body>

</html>"
`;

exports[`404 renders 1`] = `"404 - Not Found"`;
